rust/agama-server/src/software/web.rs

```diff
@@ -61,8 +67,24 @@ impl IntoResponse for SoftwareError {
 /// * `connection`: D-Bus connection to listen for events.
 pub async fn software_stream(dbus: zbus::Connection) -> Result<impl Stream<Item = Event>, Error> {
     Ok(StreamExt::merge(
-        product_changed_stream(dbus.clone()).await?,
-        patterns_changed_stream(dbus.clone()).await?,
+        StreamExt::merge(
+            StreamExt::merge(
+                product_changed_stream(dbus.clone()).await?,
+                patterns_changed_stream(dbus.clone()).await?,
+            ),
+            service_status_stream(
+                dbus.clone(),
+                "org.opensuse.Agama.Software1",
+                "/org/opensuse/Agama/Software1",
+            )
+            .await?,
+        ),
+        progress_stream(
+            dbus,
+            "org.opensuse.Agama.Software1",
+            "/org/opensuse/Agama/Software1",
+        )
+        .await,
```

Comment:
well, tokio explicitelly mention to not do multiple merge. See documentation at https://dtantsur.github.io/rust-openstack/tokio/stream/trait.StreamExt.html#method.merge
They suggest to use StreamMap for such case https://docs.rs/tokio-stream/latest/tokio_stream/struct.StreamMap.html

Reply:
Yes, I want to search for an alternative. Thanks for the hint, StreamMap looks good enough. I was also checking the [StreamExt of the futures crate](https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html).